TestProject/tests/input/RandomTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Tests\Input;
+
+use GameOfLife\Options\Getopt;
+use GameOfLife\CellularAutomat\Board;
+use GameOfLife\Input\Random;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Tests whether \Input\Random works as expected.
+ */
+class RandomTest extends TestCase
+{
+    protected $randomInput;
+    protected $getOption;
+
+    /**
+     * Initialization of the Random class.
+     *
+     * \options\Getopt will be mocked.
+     *
+     * @return void
+     */
+    protected function setUp(): void
+    {
+        $this->randomInput = new Random();
+        $this->getOption = $this->createMock(Getopt::class);
+    }
+
+    /**
+     * Initialize an empty board -> cells are all set on 0.
+     *
+     * @param $_board
+     * @return bool
+     */
+    private function emptyBoard($_board)
+    {
+        $boardIsEmpty = true;
+        foreach ($_board as $row)
+        {
+            foreach ($row as $cell)
+            {
+                if ($cell != 0)
+                    $boardIsEmpty = false;
+            }
+        }
+        return $boardIsEmpty;
+    }
+
+    /**
+     * Asserts that the board, with the random initialised living-cells, is not empty.
+     *
+     * @test
+     */
+    public function prepareRandomInput()
+    {
+        $board = new Board(10, 10);
+
+        $this->randomInput->fillBoard($board, $this->getOption);
+
+        $this->assertNotEmpty($this->emptyBoard($board->getBoard()));
+    }
+
+    /**
+     * Asserts that the board, with a defined filling-level that sets the random-chosen cells on true, is not empty.
+     *
+     * @test
+     */
+    public function fillBoardAndSetFillGrade()
+    {
+        $board = new Board(10, 10);
+        $this->getOption->method("addOptions")
+            ->with("fillingLevel")
+            ->willReturn("50");
+
+        $this->randomInput->fillBoard($board, $this->getOption);
+
+        $this->assertNotEmpty($this->emptyBoard($board->getBoard()));
```

Comment:
`assertNotEmpty()` does not make sense to check a boolean!
If you want to really check if the filling level works as expected, you must loop over all fields of the board, count all living cells and check if it more ore less equals the percentage that you gave in as parameter. For example the calculated, real percentage should be 49,50 or 51 when the fillingLevel is set to 50.